temba/triggers/models.py

```diff
@@ -271,7 +275,11 @@ def catch_triggers(cls, entity, trigger_type, channel, referrer_id=None, extra=N
             triggers = triggers.filter(models.Q(channel=channel) | models.Q(channel=None))
 
         if referrer_id is not None:
-            triggers = triggers.filter(referrer_id=referrer_id)
+            triggers = triggers.filter(models.Q(referrer_id=referrer_id) | models.Q(referrer_id=''))
```

Comment:
Realize this existed before but perhaps we should make the matching here be case insensitive, as that is probably confusing to users if we aren't matching just because of case.